src/app.js

```diff
@@ -1,13 +1,263 @@
+
+const RESERVATION_FIELDS = [ 'tripID', 'name', 'age', 'email']
+
+//************************** Render ***************
+//I should not have done all the fetching in one step.  If I were to
+// go back and fix this, I would only render the "about" details if
+// a click event happened on that row.  I would get the id for the
+// fetch either by hanging it off the html somewhere (the less-good choice)
+// or figure out what Dan was talking about when he described making a closure
+// within the tripList.forEach loop that would make an event handler for each row.
+const render = function render(tripList) {
+  $('#trip-list').html('')
+  console.log(tripList);
+  tripList.forEach((trip) => {
+    let currentTrip = new Trip({id: trip.id});
+    currentTrip.fetch().done(function() {
+      console.log(currentTrip.attributes);
+      const tripTemplate =  _.template($('#trip-template').html());
+      const tripHTML = tripTemplate(currentTrip.attributes);
+
+      $('#trip-list').append(tripHTML);
+      let id = currentTrip.get('id')
+      console.log(`.current-trip-${id}`);
+
+      $('th.sort').removeClass('current-sort-field');
+      $(`th.sort.${ tripList.comparator }`).addClass('current-sort-field');
+    });
+  });
+};
+
+//*************METHODS TO MAKE FORMS POP UP AND LEAVE AND UPDATE***************
+// toggles trip about information
+const tripDetailHandler = function (event){
+  console.log(event.target.parentElement.id);
+  let id = event.target.parentElement.id;
+  $(`.details-${id}`).toggleClass('hide');
+  $(`.button-${id}`).toggleClass('hide');
+}
+
+const modalOpener= function modalOpener() {
+  // console.log(event)
+  console.log('opening modal')
+  $('.modal').removeClass('hide');
+  $('#close').on('click', modalCloser)
+}
+
+const modalCloser= function modalCloser() {
+  // console.log(event)
+  console.log('closing modal')
+  $('.modal').addClass('hide');
+}
+
+// shows the reservation form and updates trip name
+const reserveFormUpdate = function (event){
+  console.log(event.target.parentElement.className);
+  let id = event.target.parentElement.className
+
+  let currentTrip = new Trip({id: id});
+  currentTrip.fetch().done(function() {
+    const formTemplate =  _.template($('#form-template').html());
+    console.log(formTemplate);
+    const formHTML = formTemplate(currentTrip.attributes);
+    $('#form-container').html(formHTML);
+    $('#reserve-form').on('click','.submit-reservation', addReservationHandler);
+  });
+}
+
+//**************************ADDING TRIPS AND RESERVATIONS ***************
+const addTripHandler = function(event){
+  event.preventDefault();
+  $('.trip-status-messages').html('<p> </p>')
+  const trip = new Trip(readTripFormData());
+
+  if (!trip.isValid()) {
+    handleValidationFailuresTrip(trip.validationError);
+    return;
+  }
+  console.log('I am getting ready to make a trip.')
+  console.log(trip.attributes)
+  tripList.add(trip);
```

Comment:
Note that **nothing** makes the list re-render when you add this trip, if you want it to show up on the list you need to either call render or better yet add:

`tripList.on('update', render);`

in $(document).ready